packages/strapi-design-system/src/v2/LinkButton/LinkButton.tsx

```diff
@@ -42,11 +42,23 @@ const LinkWrapper = styled(BaseButtonWrapper)`
 
 export const LinkButton = React.forwardRef<HTMLAnchorElement, LinkButtonProps>(
```

Comment:
The current implementation of the Flex and Box components is currently limited due to the utilization of the shouldForwardProp function from styled-components. While it functions well for the majority of use cases, it can pose challenges when we utilize the as property to bind a different component for rendering in the DOM, as the necessary props are not forwarded to the component.

To illustrate this behavior, [here's a brief demonstration](https://codesandbox.io/s/styled-components-forked-ymdeqz?file=/index.js).